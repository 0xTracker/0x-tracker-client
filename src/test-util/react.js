import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import React from 'react';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

export { renderWithRouter };
