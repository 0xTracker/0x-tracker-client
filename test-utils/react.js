import { createMemoryHistory } from 'history';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from 'react-testing-library';
import { Router } from 'react-router-dom';
import React from 'react';

import reducer from '../src/redux/reducer';

const withRedux = (
  component,
  { initialState, store = createStore(reducer, initialState) } = {},
) => ({
  component: <Provider store={store}>{component}</Provider>,
  store,
});

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

function renderWithRedux(component, options) {
  const wrappedComponent = withRedux(component, options);

  return {
    ...render(wrappedComponent.component),
    store: wrappedComponent.store,
  };
}

const renderWithAppContext = (
  component,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState,
    store = createStore(reducer, initialState),
  } = {},
) => ({
  ...render(
    <Provider store={store}>
      <Router history={history}>{component}</Router>
    </Provider>,
  ),
  history,
});

export { renderWithAppContext, renderWithRedux, renderWithRouter, withRedux };
