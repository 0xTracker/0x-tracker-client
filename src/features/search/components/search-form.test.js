import React from 'react';

import SearchForm from './search-form';
import { renderWithRouter } from '../../../../test-utils/react';

it('should render without props', () => {
  const { container } = renderWithRouter(<SearchForm />);

  expect(container.firstChild).toMatchSnapshot();
});
