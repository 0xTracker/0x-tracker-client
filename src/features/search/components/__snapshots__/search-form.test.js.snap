// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without props 1`] = `
<form
  class="form-inline"
  method="get"
>
  <input
    aria-label="Search"
    class="form-control input_b3fn8f"
    placeholder="Order Hash / Tx Hash / Maker / Taker"
    required=""
    type="search"
    value=""
  />
  <button
    class="btn btn-secondary btn-icon button_aompql"
    type="submit"
  >
    <svg
      fill="currentColor"
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style="vertical-align: middle;"
      viewBox="0 0 40 40"
      width="1em"
    >
      <g>
        <path
          d="m34.8 30.2c0.3 0.3 0.3 0.8 0 1.1l-3.4 3.5c-0.1 0.1-0.4 0.2-0.6 0.2s-0.4-0.1-0.6-0.2l-6.5-6.8c-2 1.2-4.1 1.8-6.3 1.8-6.8 0-12.4-5.5-12.4-12.4s5.6-12.4 12.4-12.4 12.4 5.5 12.4 12.4c0 2.1-0.6 4.2-1.7 6.1z m-17.4-20.4c-4.1 0-7.6 3.4-7.6 7.6s3.5 7.6 7.6 7.6 7.5-3.4 7.5-7.6-3.3-7.6-7.5-7.6z"
        />
      </g>
    </svg>
  </button>
</form>
`;
